test/unit/top-app-bar/index.test.tsx

```diff
@@ -98,48 +130,49 @@ test('has correct prominent dense class', () => {
   assert.isTrue(wrapper.hasClass('mdc-top-app-bar--prominent'));
 });
 
-test('navigationIcon is rendered with navigation icon class', () => {
-  const wrapper = mount(
-    <TopAppBar navigationIcon={<RippledNavigationIcon />} />
-  );
-  assert.isTrue(
-    wrapper
-      .find('.test-top-app-bar-nav-icon')
-      .hasClass('mdc-top-app-bar__navigation-icon')
-  );
+test('top app bar style should be set by state', () => {
+  const wrapper = mount(<TopAppBar />);
+  wrapper.setState({style: {color: 'blue'}});
+  assert.equal(coerceForTesting<HTMLElement>(wrapper.getDOMNode()).style.color, 'blue');
 });
 
-test('navigationIcon is rendered as custom component that accepts a className prop', () => {
-  const wrapper = mount(
-    <TopAppBar navigationIcon={<RippledNavigationIcon />} />
-  );
-  const navigationIcon = wrapper.find(RippledNavigationIcon);
-  assert.isTrue(navigationIcon.hasClass('mdc-top-app-bar__navigation-icon'));
-});
+test('#componetDIdMount will set state scrollTarget if prop.scrollTarget exists', () => {
+  const wrapper = mount<TopAppBarWithScroll>(<TopAppBarWithScroll withRef />);
+  const topAppBar: TopAppBar = wrapper.find('TopAppBar').instance() as TopAppBar;
```

Comment:
for these casts to `as TopAppBar` can you use the coereceForTesting method?

Reply:
yes I was able to use coereceForTesting.